mts/platform/deviceinfo/deviceinfoprovider.cpp

```diff
@@ -44,27 +45,21 @@ using namespace meegomtp1dot0;
 DeviceInfoProvider::DeviceInfoProvider()
     : m_batteryStatus(new BatteryStatus(this))
 {
-    QDeviceInfo di;
-
+    DeviceInfo deviceInfo;
     SsuDeviceInfo ssuDeviceInfo;
 
     m_serialNo = ssuDeviceInfo.deviceUid();
-
-    m_deviceVersion = QString("%1 HW: %2").arg(di.version(QDeviceInfo::Os))
-                      .arg(di.version(QDeviceInfo::Firmware));
-
-    m_manufacturer = di.manufacturer().isEmpty() ? m_manufacturer : di.manufacturer();
-    m_model = di.model().isEmpty() ? m_model : di.model();
+    m_deviceVersion = QString("%1 HW: %2").arg(deviceInfo.osVersion()).arg(deviceInfo.firmwareVersion());
+    m_manufacturer = deviceInfo.manufacturer();
+    m_model = deviceInfo.prettyName();
 
     connect(m_batteryStatus, &BatteryStatus::chargePercentageChanged,
             this, &DeviceInfoProvider::onBatteryPercentageChanged);
 
     if (m_newConfigFileWasCreated) {
```

Comment:
The code path leading here, i.e. MtpDeviceInfo doing some file copies, has a bit of wtf feel on a quick glance, but then again that's already there. Just noting.

Reply:
yup...